core/node/eth_sender/src/eth_tx_aggregator.rs

```diff
@@ -382,6 +464,34 @@ impl EthTxAggregator {
         Ok(vk_hash)
     }
 
+    /// Returns whether there is a pending gateway upgrade.
+    /// During gateway upgrade, the signature of the `executeBatches` function on `ValidatorTimelock` will change.
+    /// This means that transactions that were created before the upgrade but were sent right after it
+    /// will fail, which we want to avoid.
+    async fn is_pending_gateway_upgrade(
+        storage: &mut Connection<'_, Core>,
+        chain_protocol_version: ProtocolVersionId,
+    ) -> bool {
+        // If the gateway protocol version is present in the DB, and its timestamp is larger than `now`, it means that
+        // the upgrade process on the server has begun.
+        // However, if the protocol version on the contract is lower than the `gateway_upgrade`, it means that the upgrade has
+        // not yet completed.
+        let Some(gateway_protocol_version) = storage
+            .protocol_versions_dal()
+            .get_protocol_version_with_latest_patch(ProtocolVersionId::gateway_upgrade())
```

Comment:
Maybe let's better use `pending_protocol_version` from `blocks_dal()` that reads protocol version of the last sealed l2 block? L2 block with new protocol version will be sealed 1 second (l2 block deadline) after upgrade timestamp is reached and previous batch is sealed.